crates/rbuilder/src/building/builders/block_building_helper.rs

```diff
@@ -256,6 +259,11 @@ impl<DB: Database + Clone + 'static> BlockBuildingHelperFromDB<DB> {
         self.built_block_trace.true_bid_value = true_value;
         Ok(())
     }
+
+    /// Updates the cached reads for the block state.
+    pub fn update_cached_reads(&mut self, cached_reads: CachedReads) {
+        self.block_state = self.block_state.clone().with_cached_reads(cached_reads);
+    }
```

Comment:
Why is this here as a pub function and not implemented directly on impl BlockBuildingHelper for BlockBuildingHelperFromDB?

Reply:
Sorry, my bad, just fixed it.